task/04-arrays-tasks.js

```diff
@@ -9,257 +9,256 @@
 function insertItem(arr, item, index) {
-   throw new Error('Not implemented');
+  return arr.splice(index, 0, item);
 }
 
 /**
  * Returns the n first items of the specified array
- * 
+ *
  * @param {array} arr
- * @param {number} n 
- * 
+ * @param {number} n
+ *
  * @example
  *    [ 1, 3, 4, 5 ], 2  => [ 1, 2 ]
  *    [ 'a', 'b', 'c', 'd'], 3  => [ 'a', 'b', 'c' ]
  */
 function getHead(arr, n) {
-   throw new Error('Not implemented');
+  return arr.slice(0, n);
 }
 
-
 /**
  * Returns the n last items of the specified array
- * 
+ *
  * @param {array} arr
- * @param {number} n 
- * 
+ * @param {number} n
+ *
  * @example
  *    [ 1, 3, 4, 5 ], 2  => [ 4, 5 ]
  *    [ 'a', 'b', 'c', 'd'], 3  => [ 'b', 'c', 'd' ]
  */
 function getTail(arr, n) {
-   throw new Error('Not implemented');
+  return arr.splice(-n);
 }
 
-
 /**
  * Returns CSV represebtation of two-dimentional numeric array.
  * https://en.wikipedia.org/wiki/Comma-separated_values
- * 
+ *
  * @param {array} arr
  * @return {string}
- * 
+ *
  * @example
  *    [
  *       [  0, 1, 2, 3, 4 ],
  *       [ 10,11,12,13,14 ],
  *       [ 20,21,22,23,24 ],
  *       [ 30,31,32,33,34 ]
- *    ] 
- *           => 
+ *    ]
+ *           =>
  *     '0,1,2,3,4\n'
  *    +'10,11,12,13,14\n'
  *    +'20,21,22,23,24\n'
  *    +'30,31,32,33,34'
  */
 function toCsvText(arr) {
-   throw new Error('Not implemented');
+  return arr.join(`\n`);
 }
 
 /**
  * Transforms the numeric array into the according array of squares:
  *   f(x) = x * x
- * 
+ *
  * @param {array} arr
  * @return {array}
- * 
+ *
  * @example
  *   [ 0, 1, 2, 3, 4, 5 ] => [ 0, 1, 4, 9, 16, 25 ]
  *   [ 10, 100, -1 ]      => [ 100, 10000, 1 ]
  */
 function toArrayOfSquares(arr) {
-   throw new Error('Not implemented');
+  return arr.map(element => element * element);
 }
 
-
 /**
  * Transforms the numeric array to the according moving sum array:
- *     f[n] = x[0] + x[1] + x[2] +...+ x[n] 
+ *     f[n] = x[0] + x[1] + x[2] +...+ x[n]
  *  or f[n] = f[n-1] + x[n]
- * 
+ *
  * @param {array} arr
  * @return {array}
- * 
+ *
  * Example :
  *   [ 1, 1, 1, 1, 1 ]        => [ 1, 2, 3, 4, 5 ]
  *   [ 10, -10, 10, -10, 10 ] => [ 10, 0, 10, 0, 10 ]
- *   [ 0, 0, 0, 0, 0]         => [ 0, 0, 0, 0, 0] 
+ *   [ 0, 0, 0, 0, 0]         => [ 0, 0, 0, 0, 0]
  *   [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 ] => [ 1, 3, 6, 10, 15, 21, 28, 36, 45, 55 ]
  */
 function getMovingSum(arr) {
-   throw new Error('Not implemented');
+  let resultArr = [];
+  arr.reduce(
+    (previous, current, index) => (resultArr[index] = previous + current),
+    0
+  );
+  return resultArr;
 }
 
 /**
  * Returns every second item from the specified array:
- * 
+ *
  * @param {array} arr
  * @return {array}
- * 
+ *
  * Example :
  * [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 ] => [ 2, 4, 6, 8, 10 ]
  * [ 'a', 'b', 'c' , null ]  => [ "b", null ]
  * [ "a" ] => []
  */
 function getSecondItems(arr) {
-   throw new Error('Not implemented');
+  return arr.filter((_, index) => index & 1);
```

Comment:
This solution works but sometimes it's better to use bigger but more readable solution instead of the shorter one. In this case it can be just `index % 2`

Reply:
Not so many people use bitwise operators so I think it should be a big reason to use it